code/addons/test/src/vitest-plugin/index.ts

```diff
@@ -109,14 +112,27 @@ export const storybookTest = async (options?: UserOptions): Promise<Plugin> => {
     getStoryGlobsAndFiles(presets, directories),
     presets.apply('framework', undefined),
     presets.apply('env', {}),
-    presets.apply('viteFinal', {}),
+    presets.apply<{ plugins?: Plugin[] }>('viteFinal', {}),
     presets.apply('staticDirs', []),
     extractTagsFromPreview(finalOptions.configDir),
   ]);
 
-  return {
+  // filter out plugins that we know are unnecesary for tests, eg. docgen plugins
+  const plugins = (await withoutVitePlugins(
+    (viteConfigFromStorybook.plugins as unknown as PluginOption[]) ?? [],
+    [
+      'storybook:package-deduplication', // addon-docs
+      'storybook:mdx-plugin', // addon-docs
+      'storybook:react-docgen-plugin',
+      'vite:react-docgen-typescript', // aka @joshwooding/vite-plugin-react-docgen-typescript
+      'storybook:svelte-docgen-plugin',
+      'storybook:vue-component-meta-plugin',
+      'storybook:vue-docgen-plugin',
```

Comment:
Do we also have to add all the Next.js related vite plugins?

- `vite-plugin-storybook-nextjs`
- `vite-plugin-storybook-nextjs-swc`
- `vite-plugin-next-mocks` (damn it, not consistent with the naming of the other plugins :/)
- `vite-plugin-storybook-nextjs-image`
- `vite-plugin-storybook-nextjs-font`
- `vite-plugin-storybook-nextjs-env`
- `vite-plugin-storybook-nextjs-dynamic`

Source: https://github.com/storybookjs/vite-plugin-storybook-nextjs/blob/main/src/index.ts#L39

Reply:
no we're only filtering out plugins that are unnecessary for tests, ie. plugins that don't impact the story rendering - see comment at L120.

The plugins you list there would most definitely make stories fail if they weren't included.

Reply:
Sorry. My mistake! Then all good!